graphs/find_connected_components.cpp

```diff
@@ -0,0 +1,65 @@
+#include "../headers.h"
+
+/* Get connected components in a graph */
+
+void get_connected_components(vector<vi>& g, int n)
+{
+	/* Creating the graph */
+	map<int, set<int>> graph;
+	for(auto c: g)
+	{
+		graph[c[0]].insert(c[1]);
+		graph[c[1]].insert(c[0]);
+	}
+
+	vector<set<int>> result;
+	std::vector<int> visited(n, false);
+	int components = 0;
+
+	for(int i=0; i<n; i++)
+	{
+		set<int> path = {};
+
+		if(visited[i])
+			continue;
+		else
+			components += 1;
+
+		deque<int> stack = {i};
```

Comment:
consider using stack, don't see a need for deque. All the functions, you are doing with deque can be done with stack

Reply:
Then why stack? What makes it better than deque?

Reply:
Stack will internally create deque by default, it will make the code more readable and less error prone. using a wrapper is better.